app/core/api.py

```diff
@@ -62,3 +62,47 @@ async def execute(request: ExecuteRequest) -> list[str]:
         raise HTTPException(status_code=400, detail=str(e))
     except Exception as e:
         raise HTTPException(status_code=500, detail=str(e))
+
+
+class MemCleanRequest(BaseModel):
+    """Request model for agent memory cleanup endpoint.
+
+    Attributes:
+        aid (str): Agent ID to clean
+        thread_id (str): Thread ID to clean
+    """
+
+    aid: str
+    thread_id: str | None = Field("")
+
+
+@core_router.post("/memclean")
```

Comment:
please move to app/admin, and better start with /agents, then it will be exposed. don't forget to use verify_jwt, you can use your jwt call all /agents API.